src/core/gui/XournalppCursor.cpp

```diff
@@ -515,6 +520,44 @@ auto XournalppCursor::createHighlighterOrPenCursor(int size, double alpha) -> Gd
 }
 
 
+auto XournalppCursor::createHorizontalLineCursor(int size, double alpha, GdkWindow* theWindow) -> GdkCursor* {
+    gint x, y;
+    GdkDevice *mouse_device;
+
+    #if GTK_CHECK_VERSION (3,20,0)
+    GdkSeat *seat = gdk_display_get_default_seat (gdk_display_get_default ());
+    mouse_device = gdk_seat_get_pointer (seat);
+    #else
+    GdkDeviceManager *devman = gdk_display_get_device_manager (gdk_display_get_default ());
+    mouse_device = gdk_device_manager_get_client_pointer (devman);
+    #endif
+
+    gdk_window_get_device_position (theWindow, mouse_device, &x, &y, NULL);
+
+    int height = size;
+    int width = gdk_window_get_width(theWindow);
```

Comment:
That only seems to work if all pages are fully displayed in the window. Otherwise the cursor gets too large. Just zoom in or add some pages (displayed in 1 row).

Reply:
Oof, yeah, you're right. I need to get the width of just that main view, or maybe the width of the visible pixels? I see that there's a way to get the visible document coordinates... can that be somehow converted to screen pixels?

Reply:
I also tried getting the zoom factor, but I'm afraid those calculations may be too unreliable.

Reply:
Yep, I think the best way to do this would be if I could somehow get the coordinates of the screen that the currently active page takes up and draw a line across _just that_ at the coords of the cursor. I'm like... 40% sure that's possible.

Reply:
Getting there. I need to figure out if I can somehow clamp my cairo draw starting point at the edge of the window, or the edge of the sidebar.

Reply:
I think I need to find the size of the sidebar as well....

Reply:
Have you considered using `getVisibleRect`?